src/Enum.php

```diff
@@ -46,7 +46,7 @@ abstract class Enum implements \JsonSerializable
      * @psalm-param static<T>|T $value
      * @throws \UnexpectedValueException if incompatible type is given.
      */
-    public function __construct($value)
+    final public function __construct($value)
```

Comment:
This fixes for the `Psalm UnsafeInstantiation` and the reference is available [here](https://psalm.dev/docs/running_psalm/issues/UnsafeInstantiation/).

Reply:
Fix following issues:

```
ERROR: UnsafeInstantiation - src/Enum.php:138:29 - Cannot safely instantiate class MyCLabs\Enum\Enum with "new static" as its constructor might change in child classes (see https://psalm.dev/229)
            $values[$key] = new static($value);

ERROR: UnsafeInstantiation - src/Enum.php:221:20 - Cannot safely instantiate class MyCLabs\Enum\Enum with "new static" as its constructor might change in child classes (see https://psalm.dev/229)
            return new static($array[$name]);

```

Reply:
Fixing above issues has two approaches and we can choose of them to fix and they're as follows:

- Let class constructor be `final` and it can avoid extended classes modify.
- Adding the `@psalm-consistent-constructor` comment annotation on class comment annotations.

I'm not sure which it's good/preferred to fix this issue so I choose using `final` for the class constructor.

Reply:
This is a BC break so we can't switch to final though

Reply:
Thanks. We can use second approach about adding the `@psalm-consistent-constructor` comment annotation to complete this issue :).

Reply:
👍